spec/recentReleaseSpec.js

```diff
@@ -0,0 +1,184 @@
+            },
+            getSteamList : function(){
+                
+                $httpBackend.expectGET(steamListUrl)
+                .respond(function(status, data, headers, statusText) {
+                    
+                    return [200, 'OK'];
+                });
+            }
+        };
+        $scope.newReleases = {data : releaseArray};
+        
+        spyOn(requestHandler,'getReleases');
+        spyOn(requestHandler,'getSteamList');
+        
+        requestHandler.getReleases();
+        requestHandler.getSteamList();
+        
+        createController = function(){
+            return $controller('recentReleasesCtrl',{'$scope' : $scope});
+        };
+        
+    }));
+    
+    it("Should be true if the request for recent releases was sent", function(){
+        
+        expect(requestHandler.getReleases).toHaveBeenCalled();
+    });
+    
+    it("Should be true if the request for the list of steam games was fulfilled", function() {
+        
+        expect(requestHandler.getSteamList).toHaveBeenCalled();
+    });
+    
+    it("Should be true if the press score and the player's score don't exist",function() {
+       
+       expect(!$scope.newReleases.data[8].aggregated_rating && !$scope.newReleases.data[8].rating).toBe(true); 
+    });
+    
+    it("Should be true if a game is on PC but does not exist on steam", function() {
+        
+    });
+    
+    it("Should be true if the press score exists", function() {
+        $httpBackend.expectGET(igdb_releases,{
+        'X-Mashape-Key': 'MY_IGDB_API_KEY',
+        'Accept' : 'application/json'
+        }).respond(function(status, data, headers, statusText) {
+ 
+            return [200,{data: releaseArray},'OK'];
+        });
+        
+        expect(!$scope.newReleases.data[1].aggregated_rating).toBe(false);
+    });
+    
+    
+    it("Should be true if the press rating >= 75", inject(function($http){
+
+        
+        expect($scope.newReleases.data[1].aggregated_rating >= 75).toBe(true);
+    }));
+    
+    it("Should be true if the press rating does not exist and the player rating >= 75", function() {
+       
+        expect(!$scope.newReleases.data[0].aggregated_rating && $scope.newReleases.data[0].rating >= 75).toBe(true); 
+    });
+    
+    it("Should be true if the press rating is < 75", function() {
+       
+       expect($scope.newReleases.data[2].aggregated_rating < 75).toBe(true); 
+    });
+    
+    
+    it("Should be true if the image ID exists", function() {
+        
+        expect(!$scope.newReleases.data[1].cover.cloudinary_id).toBe(false);
+    });
+    
+    /*
+    The following tests are for comparing names from IGDB with game names on the steam store
+    */
+    
+    it("Should replace colon trailing space with dash and spaces", function() {
+        
+        var foo = "Warlock: master of the arcane";
+        var bar = "Warlock - master of the arcane";
+        expect(foo.replace(": "," - ")).toBe(bar);
```

Comment:
you should be testing your own code, the the built in replace function. You need to mock input to your releaseNameCleaner function and then check the output, to make this easier you my have to rewrite that function.